algorithms/src/Kokkos_Random.hpp

```diff
@@ -537,15 +537,15 @@ struct rand<Generator, Kokkos::complex<double> > {
   }
 };
 
-template <class DeviceType>
+template <class DeviceType, class>
```

Comment:
Can you explain what the rational is for adding this additional template parameter?

Reply:
To be able to SFINAE on it, namely distinguish is the template argument actually is a device type.
#2693 reported that these classes actually only worked when an execution space was used instead of a device type...